File: src/lib/api/rPCTestClient.ts
import * as pb from "../pb";
import { RPCHost } from "../rpc/host";
import { Readable as GenericReadable } from "../rpc/stream";

export default class RPCTest {
  constructor(private readonly host: RPCHost) {}

  public callUnary(
    arg: pb.IRPCCallUnaryRequest = new pb.RPCCallUnaryRequest()
  ): Promise<pb.RPCCallUnaryResponse> {
    return this.host.expectOne(
      this.host.call("RPCTest/CallUnary", new pb.RPCCallUnaryRequest(arg))
    );
  }
  public callStream(
    arg: pb.IRPCCallStreamRequest = new pb.RPCCallStreamRequest()
  ): GenericReadable<pb.RPCCallStreamResponse> {
    return this.host.expectMany(
      this.host.call("RPCTest/CallStream", new pb.RPCCallStreamRequest(arg))
    );
  }
}

File: android/app/src/main/java/gg/strims/ppspp/rpc/RPCTestClient.kt
package gg.strims.ppspp.rpc
import gg.strims.ppspp.proto.*

class RPCTestClient(filepath: String) : RPCClient(filepath) {

    suspend fun callUnary(
        arg: RPCCallUnaryRequest = RPCCallUnaryRequest()
    ): RPCCallUnaryResponse =
        this.callUnary("RPCTest/CallUnary", arg)

    suspend fun callStream(
        arg: RPCCallStreamRequest = RPCCallStreamRequest()
    ): RPCResponseStream<RPCCallStreamResponse> =
        this.callStreaming("RPCTest/CallStream", arg)

}
